From ef329b30b80d81254f775259f00679dae02bfa40 Mon Sep 17 00:00:00 2001
From: Brad Love <hidden@email.co>
Date: Thu, 16 Mar 2017 23:13:05 +0000
Subject: [PATCH 03/28] em28xx: usb packet size tweaks

Hauppauge em28xx bulk devices exhibit continuity errors and missing/corrupted
packets, when run in VMWare virtual machines. Unknown if other manufacturers
bulk models exhibit the same issue. KVM/Qemu is unaffected.

This changes the size of bulk transfers to have a beneficial alignment.
Before:
# 512 * 384 = 196608
## 196608 % 188 != 0

After:
# 512 * 47 * 8 = 192512    (188 * 128 * 8)
## 192512 % 188 = 0

Next common multiple is 216576, keeping it under the original max
for backwards compatibility.

Successful usage under load afterwards natively and in both VMWare
and KVM/Qemu virtual machines.
---
 drivers/media/usb/em28xx/em28xx.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/usb/em28xx/em28xx.h b/drivers/media/usb/em28xx/em28xx.h
index c85292c9883b..7542f33c1b11 100644
--- a/drivers/media/usb/em28xx/em28xx.h
+++ b/drivers/media/usb/em28xx/em28xx.h
@@ -191,7 +191,7 @@
    USB 2.0 spec says bulk packet size is always 512 bytes
  */
 #define EM28XX_BULK_PACKET_MULTIPLIER 384
-#define EM28XX_DVB_BULK_PACKET_MULTIPLIER 384
+#define EM28XX_DVB_BULK_PACKET_MULTIPLIER 376   /* mod 47 for 188 alignment */
 
 #define EM28XX_INTERLACED_DEFAULT 1
 
-- 
2.11.0

